subsys/usb/class/dfu/usb_dfu.c

```diff
@@ -344,13 +321,61 @@ static void dfu_reset_counters(void)
 {
 	dfu_data.bytes_sent = 0U;
 	dfu_data.block_nr = 0U;
-	if (flash_img_init(&dfu_data.ctx)) {
-		LOG_ERR("flash img init error");
-		dfu_data.state = dfuERROR;
-		dfu_data.status = errUNKNOWN;
+}
+
+#define READONLY(part) \
+	[DT_FIXED_PARTITION_ID(part)] = DT_PROP(part, read_only),
+
+#define READONLY_FOREACH_PARTITION(n) \
+	DT_FOREACH_CHILD(DT_DRV_INST(n), READONLY)
+
+/**
+ * @brief Helper function to check if given partition is read-only
```

Comment:
I see no benefit in using Doxygen for internal functions.

Reply:
Some (not all) internal helper functions in this file had Doxygen comments, should they be added for new helper functions then or not?

Reply:
Please not for internal functions, I do not know why people do it.